application-glossary-deepl/src/main/resources/Glossary/Code/DeepL/GlossaryDeepLService.xml

```diff
@@ -0,0 +1,317 @@
+import org.xwiki.model.reference.DocumentReference
+import org.xwiki.model.reference.ObjectReference
+import org.xwiki.query.Query
+import org.xwiki.query.QueryManager
+import org.xwiki.script.service.ScriptService
+import org.xwiki.script.service.ScriptServiceManager
+
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Provider
+import javax.inject.Singleton;
+
+@Component
+@Named("glossary.deepl")
+@Singleton
+public class DeepLGlossaryScriptService implements ScriptService
+{
+    @Inject
+    private Logger logger
+
+    @Inject
+    private QueryManager queryManager
+
+    @Inject
+    protected Provider&lt;XWikiContext&gt; xwikiContextProvider
+
+    @Inject
+    private ComponentManager componentManager
+
+    @Inject
+    private ScriptServiceManager scriptServiceManager
+
+    private Translator translator = null
+    // Helper
+    private static String getDeepLLanguage(String language)
+    {
+        return language.split('_')[0]
+    }
+
+    private Translator getTranslator()
+    {
+        def deeplScriptService = scriptServiceManager.get("translator.deepl")
+        try {
+            if (translator == null) {
+                String apiKey = deeplScriptService.getApiKey()
+                translator = new Translator(apiKey)
+            }
+        } catch (XWikiException e) {
+            logger.error("Error while retrieving API key.", e)
+            return null
+        }
+        return translator
+    }
+
+    private static boolean supportsLanguageCombination(List&lt;GlossaryLanguagePair&gt; deeplSupportedGlossaries,
+        String sourceLanguage, String targetLanguage)
+    {
+        for (entry in deeplSupportedGlossaries) {
+            if (entry.sourceLanguage == getDeepLLanguage(sourceLanguage) &amp;&amp; entry.targetLanguage ==
+                getDeepLLanguage(targetLanguage))
+            {
+                return true
+            }
+        }
+        return false
+    }
+
+    private static List&lt;String&gt; getGlossariesWithName(List&lt;GlossaryInfo&gt; deeplGlossaries, String glossaryName)
+    {
+        List&lt;String&gt; glossaryIds = []
+        deeplGlossaries.each { glossary -&gt;
+            if (glossary.name == glossaryName) {
+                glossaryIds.add(glossary.glossaryId)
+            }
+        }
+        return glossaryIds
+    }
+
+    private String GetGlossaryNamePrefix()
+    {
+        ConfigurationSource xwikiPropertiesConfiguration =
+            componentManager.getInstance(ConfigurationSource.class, 'xwikiproperties')
+        String glossariesPrefix = xwikiPropertiesConfiguration.getProperty('deepl.glossaries.prefix', '')
+        def wikiPrefix = xwikiContextProvider.get().getWikiId()
+
+        if (StringUtils.isBlank(glossariesPrefix)) {
+            return "${wikiPrefix}"
+        } else {
+            return "${glossariesPrefix}-${wikiPrefix}"
+        }
+    }
+    // Method used by different pages
+    def SynchronizeGlossaries()
```

Comment:
What about permissions? Should any user with script right be able to write a script that calls this method?